### PR TITLE
Add ULDM preset model

### PR DESCRIPTION
**Pull request by AlexLaroche7**
- **Added 'ULDM' preset model to preset_models.** Uses WDMGeneral skeleton. Characteristic mass at which point CDM and ULDM differ and cutoff mass are both calculated within the function, can be changed depending on how you want to format it. The cutoff mass is given as a lambda function. I noticed that inputting a function for 'log_mlow' substantially increases runtime, but only up to a few minutes. ULDM particle mass and ULDM core-halo power law exponent are also arguments for the preset model.
- **Included Zeta function in 'ULDM' halo model.** This is relevant for the redshift dependence of the core radius. Important when creating halos near the cutoff mass.
- Minor change to test_uldm, can ignore.

### CODE REVIEW

File: pyHalo/Halos/HaloModels/ULDM.py [hunk 1]
```diff
@@ -159,14 +161,17 @@ def _rescaled_tnfw_params(self, tnfw_params, uldm_params):
...
             raise ValueError('The resulting composite NFW profile mass is negative, tweak your parameters, factor = ' + str(factor)
                             + ', M_uldm = ' + str(M_uldm) + ', M_nfw = ' + str(M_nfw))
         else:
             pass
 
         tnfw_params['alpha_Rs'] *= factor
         return tnfw_params
+    
+    def _zeta(self,z):
+        Om_z = Cosmology().astropy.Om(z)
```
Comment (dangilman) [inline pyHalo/Halos/HaloModels/ULDM.py:173]: Can you use the instance of cosmology that exists in the instance of LensCosmo (_lens_cosmo_instance) just to make sure the same cosmological parameters are being used throughout? just self._lens_cosmo_instance.cosmo

File: pyHalo/preset_models.py [hunk 2]
```diff
@@ -239,4 +241,122 @@ def SIDM(z_lens, z_source, cross_section_name, cross_section_class, kwargs_cross
...
+
+    # this will use the default cosmology. parameters can be found in defaults.py
+    pyhalo = pyHalo(z_lens, z_source)
+    # Using the render method will result a list of realizations
+    realization_subs = pyhalo.render(['SUBHALOS'], kwargs_model_subhalos, nrealizations=1)[0]
+    realization_line_of_sight = pyhalo.render(['LINE_OF_SIGHT', 'TWO_HALO'], kwargs_model_field, nrealizations=1)[0]
+
+    uldm_realization = realization_line_of_sight.join(realization_subs, join_rendering_classes=True)
+
+    return uldm_realization
```
Comment (dangilman) [inline pyHalo/preset_models.py:361]: can you add a test function for this function inside the test_preset_models file to make just it executes without throwing an error?

**Review by dangilman — CHANGES_REQUESTED** [review]
Hi Alex, looks good to me! I have two very minor comments

File: pyHalo/Halos/HaloModels/ULDM.py (hunk 1 above)
Comment (AlexLaroche7) [reply pyHalo/Halos/HaloModels/ULDM.py]: Fixed

File: pyHalo/preset_models.py (hunk 2 above)
Comment (AlexLaroche7) [reply pyHalo/preset_models.py:361]: Fixed

**Comment by AlexLaroche7** [comment]
That should address both issues. Let me know if there is anything else, thanks for the feedback!

File: pyHalo/preset_models.py (hunk 2 above)
Comment (AlexLaroche7) [reply pyHalo/preset_models.py:361]: Never mind, called the function incorrectly, will fix shortly.

**Pull request merged**